diff --git a/src/polyfills/require.js b/src/polyfills/require.js
index 2dc0c618aae0d3f6d21dd7991ccc180fee879891..5c255a56e2c9678703ef0d9601aeae84ba660d25 100644
--- a/src/polyfills/require.js
+++ b/src/polyfills/require.js
@@ -89,8 +89,10 @@ function metroRequire(moduleId) {
     : guardedLoadModule(moduleIdReallyIsNumber, module);
 }
 function shouldPrintRequireCycle(modules) {
-  const regExps =
-    global[__METRO_GLOBAL_PREFIX__ + "__requireCycleIgnorePatterns"];
+  const regExps = global[
+    __METRO_GLOBAL_PREFIX__ + "__requireCycleIgnorePatterns"
+  ] || // HACK: Expo SSR mode doesn't have this global. Fall back to the default.
+  [/(^|\/|\\)node_modules($|\/|\\)/];
   if (!Array.isArray(regExps)) {
     return true;
   }
