diff --git a/build/vendor/vercel.js b/build/vendor/vercel.js
index bf59a827ea24e6a27b258dfe9142c259c9061a7b..436018784cb9005b7f8f91108749319bf670e054 100644
--- a/build/vendor/vercel.js
+++ b/build/vendor/vercel.js
@@ -59,7 +59,7 @@ function convertRequest(req, res) {
 exports.convertRequest = convertRequest;
 async function respond(res, expoRes) {
     res.statusMessage = expoRes.statusText;
-    res.writeHead(expoRes.status, expoRes.statusText, [...expoRes.headers.entries()]);
+    res.writeHead(expoRes.status, expoRes.statusText, [...expoRes.headers.entries()].flat());
     if (expoRes.body) {
         await (0, node_1.writeReadableStreamToWritable)(expoRes.body, res);
     }
